shared/cmd/ask.go

```diff
@@ -14,6 +14,11 @@ import (
 
 var stdin = bufio.NewReader(os.Stdin)
 
+// SetReader sets the default reader used to interpret user input.
+func SetReader(reader *bufio.Reader) {
```

Comment:
Hrm, not sure I like this approach as its modifying global state that would affect other packages that import this.

Can we move these `Ask* function into their own struct and package, like `github.com/canonical/lxd/shared/ask` and then we can have a function like `NewAsker(reader *bufio.Reader) *Asker`?

Reply:
Good point, I'll do that now.